Tickets/preops 3615

galaxyumi opened the pull request:
Add advice about passing lists with table joins for DP0.3. Note that while subqueries are possible for DP0.3, they're not recommended because subqueries are not supported in Qserv-hosted databases.

galaxyumi committed:
Update table-access-and-queries.rst 

adding ADQL advice on passing a list to a query instead of using subqueries.

galaxyumi committed:
Update table-access-and-queries.rst

Minor edits for clarification

galaxyumi committed:
Update table-access-and-queries.rst

More edits for clarification

ameisner reviewed (changes requested):
Regarding:

> LSST Query Services (Qserv) do not support subqueries. Thus, using subqueries are not recommended although DP0.3 is not hosted on Qserv, but on the Rubin Science Platform (RSP).

I would delete the last clause of the second sentence here ", but  on the Rubin Science Platform (RSP)". I thought the alternative to Qserv being used for DP0.3 was essentially a standard PostgreSQL database setup (?), which to my understanding is a separate topic from whether or not a database is accessible via RSP.

Regarding:

> Refer [DP0.2 ADQL Recipes](https://dp0-2.lsst.io/data-access-analysis-tools/adql-recipes.html) for more general ADQL advice.

I would consider deleting this sentence. At one time I added a reference to the DP0.2 ADQL advice page from the DP0.3 documentation but then Melissa told me that the DP0.2 advice is essentially not relevant to DP0.3. At the least I would change "Refer DP0.2 ADQL Recipes" to "Refer to the DP0.2 ADQL Recipes" as a minor wording edit.

ameisner commented:
Here's the relevant comment from Melissa, during the DP0.3 sprint:

> I noticed https://dp0-3.lsst.io/v/PREOPS-3493/data-products-dp0-3/index.html#adql-recipes links to dp0-2 but that dp0-2 page has no usable advice or recipes for dp0-3.

ChristinaAdair reviewed (changes requested):
LSST Query Services (Qserv) do not support subqueries. Thus, using subqueries are **not recommended** although DP0.3 is not hosted on Qserv, but on the Rubin Science Platform (RSP). Instead, when having a list of objects in hand either from a previous query or a user-provided catalog,

Maybe rewrite the second sentence.  " Thus, using subqueries on the Rubin Science Platform (RSP) is **not recommended** ."

galaxyumi committed:
Update table-access-and-queries.rst

remove a reference to the DP0.2 ADQL advice page for general advice

galaxyumi committed:
Update table-access-and-queries.rst

removing the RSP part

galaxyumi commented:
@ameisner @ChristinaAdair Thanks for your suggestions! I made those changes.

ameisner committed:
very minor verb conjugation edit

MelissaGraham commented on data-products-dp0-3/table-access-and-queries.rst:
```diff
@@ -117,6 +118,39 @@ The N:1 nature of these joins means that the data retrieved can contain columns
+    ON dia.diaSourceId = sss.diaSourceId
+    WHERE dia.ssObjectId
+    IN {}
+    """.format(tuple(df_uniqueObj['ssObjectId']))
```
The fact that the code snippet starts with an import makes it seem like the whole thing is copy-paste-executable, but it is not because df_uniqueObj is not defined. Can you make the entire example copy-paste-executable by defining a list of a few ssObjectId that gets passed to the query, instead of df_uniqueObj?

galaxyumi replied:
Ok, let me try that.

ameisner reviewed (approved):
Looks good to me at this point (I added one tiny commit as well).

MelissaGraham commented on data-products-dp0-3/table-access-and-queries.rst:
```diff
@@ -117,6 +118,39 @@ The N:1 nature of these joins means that the data retrieved can contain columns
+    results = service.search(query).to_table()
+
+
+A relevent example for DP0.2 can be found `here <https://dp0-2.lsst.io/data-access-analysis-tools/adql-recipes.html#individual-objects>`__. 
```
Probably not needed to link to DP0.3 as the above code snippet provides a relevant example.

ameisner replied:
Thanks...I also now notice that "relevent" is misspelled in that sentence, though if the whole thing gets deleted then it's not an issue...

galaxyumi replied:
Argh... Thanks. I will delete the sentence.

galaxyumi committed:
Update table-access-and-queries.rst

remove a link to the DP0.2 document and make the code snippet fully executable.

MelissaGraham commented on data-products-dp0-3/table-access-and-queries.rst:
```diff
@@ -117,6 +118,38 @@ The N:1 nature of these joins means that the data retrieved can contain columns
 
+.. _DP0-3-Table-Access-ADQL-passing-list:
+
+Advice for passing an existing list to a new query
```
Could we shorten? E.g., to "Passing a list to a query" or "List-based query" or "Query a list of objects", or similar.

MelissaGraham commented on data-products-dp0-3/table-access-and-queries.rst:
```diff
@@ -117,6 +118,38 @@ The N:1 nature of these joins means that the data retrieved can contain columns
+previous query or a user-provided catalog, the list, formatted as a python tuple, can be passed to a 
+new query for table joins. The example query below is to retrieve information about individual observations 
+from the ``DiaSource`` and ``SSSource`` tables for indivdual unique objects selected from the ``SSObject`` 
+table and stored in a ``df_uniqueObj`` table from a previous query.
```
df_uniqueObj is referenced but not used, change that to sId_list or vice versa

MelissaGraham commented on data-products-dp0-3/table-access-and-queries.rst:
```diff
@@ -117,6 +118,38 @@ The N:1 nature of these joins means that the data retrieved can contain columns
+new query for table joins. The example query below is to retrieve information about individual observations 
+from the ``DiaSource`` and ``SSSource`` tables for indivdual unique objects selected from the ``SSObject`` 
+table and stored in a ``df_uniqueObj`` table from a previous query.
+
```
Let's add a statement about how the example uses only three objects but lists can be very long (tens of thousands? did you test how long a list could be?)

MelissaGraham commented on data-products-dp0-3/table-access-and-queries.rst:
```diff
@@ -117,6 +118,38 @@ The N:1 nature of these joins means that the data retrieved can contain columns
+.. code-block:: python
+
+    from lsst.rsp import get_tap_service, retrieve_query
+    service = get_tap_service()
```
this will not execute as is. need:
`service = get_tap_service("ssotap")`

MelissaGraham commented on data-products-dp0-3/table-access-and-queries.rst:
```diff
@@ -117,6 +118,38 @@ The N:1 nature of these joins means that the data retrieved can contain columns
+    """.format(tuple(sId_list))
+
+    results = service.search(query).to_table()
+
```
Perhaps we could add a final statement about how this returns a results table with 1915 rows, with 597, 572, and 746 rows for each of the SSObjects, respectively.

MelissaGraham reviewed (changes requested):
A few specific final suggestions. Nearly there!

RST should be written one-sentence-per-line (break after periods) and then long sentences should also be truncated, if that could be updated too.

galaxyumi committed:
Implement Melissa's feedback

galaxyumi committed:
breaking into 3 paragraphs

galaxyumi committed:
shorten the title
